src/hooks/useDiskPercentage.ts

```diff
@@ -0,0 +1,37 @@
+import { cozyFetch } from 'actions'
+import { computeDiskPercent } from 'lib/computeDiskPercent'
+import { useEffect, useState } from 'react'
+import logger from 'lib/logger'
+
+const warn = (logger as { warn: (...args: unknown[]) => void }).warn
+type ExpectedJSON = {
+  data: { attributes: { used: string; quota: string } }
+}
+
+export const useDiskPercentage = (): string | undefined => {
+  const [percent, setPercent] = useState<undefined | string>()
+
+  useEffect(() => {
+    const asyncCore = async (): Promise<void> => {
+      try {
+        const json = (await cozyFetch(
```

Comment:
You should NEVER add a code that makes a stack request without using cozy-client. useClient / client.query, useFetchJSON... you have a lof of choice.

Reply:
Like that, in the future CozyClient will do the type thing for you.

Reply:
client.query(Q('io.cozy.settings').getById('disk-usage')) for instance

Reply:
With this PR, I saw that we still have a lot of requests not using cozy-client in settings... :'(

Reply:
cozyFetch est un vieux composant, au pire si la route de la stack n'est pas prise en compte par cozy.client, on peut faire un `client.getStack().fetchJSON()` ou quelque chose comme ça

Reply:
@Crash-- any opinion of what should be the absolute best API to use in this specific case?

Reply:
const { data } = await client.query(Q('io.cozy.settings').getById('disk-usage'))
setPercent()...

Reply:
Updated